pdf/plugin-dowker/main.py

```diff
@@ -0,0 +1,165 @@
+    ft_lookup = {}
+    matrix = []
+    curr=0
+    errorMatrix = np.asmatrix(np.zeros(shape=(30,len(ft_names)+1)))
+    set1 = set()
+    set2 = set()
+    mode = 'files'
+    lines=list()
+    file1 = open("MyFile.txt", "w") 
+    for line in sys.stdin:
+        line = line.strip()
+        file1.write(line)
+
+        if not line:
+            continue
+        
+        if line == 'REFERENCES':
+            mode = 'refs'
+        elif mode == 'files':
+            obj = json.loads(line)
+            
+            file_names.append(obj.pop('_id'))
+            file_names_backward[file_names[-1]] = len(file_names) - 1
+            matrix.append({})
+          #  lines.append(line)
+            i=1
+            a="pdfinfo_<>"
+            matrix[-1][0]=file_names[-1]
+            curr+=1
+          #  errorMatrix.resize(len(ft_names),curr, False)
+            
+            for name in ft_names:
+                matrix[-1][i]=0
+                errorMatrix[curr,i]=1
+                matched =False
+                for key in obj.keys():
+                    if name.split("\\<\\<")[0]==key.split("\\<\\<")[0]:
+                        matrix[-1][i]=obj[key]
+                        errorMatrix[curr,i]=1-int(obj[key])
+                        matched=True
+                        break
+                if not(matched):
+                    errorMatrix[curr, i]=1
+           #     matrix[-1][i]="|"+list(obj.keys())[0] +"|"
+               # matrix[-1][i]=obj[list(obj.keys())[0]]
+               # if name in obj:
+                #    matrix[-1][i]=obj[name]
+               # else:
+                #    matrix[-1][i]=name
+              #  matrix[-1][i]=line[0:300]
+                i+=1
+#            for k, v in obj.items():
+ #               print(k)
+                
+        elif mode == 'refs':
+            # Ensure UI maintains its old decision (this was a regression; the
+            # new code does not have this flaw)
+            #print(line)
+            lines.append(line)
+            if not use_refs:
+                continue
+
+            obj = json.loads(line)
+            if obj['status'] == 'valid':
+                set1.add(obj['testfile'])
+            else:
+                set2.add(obj['testfile'])
+        else:
+            raise NotImplementedError(mode)
+    file1.close()
+	#Create Dowker from matrix
+    errorMatrixToDowker(errorMatrix)
+	#Return plotly args
+	
+
+    # Sort files based on likelihood of being 'odd'
+
+    html_vars['lines']=str(errorMatrix)
+    write_html(os.path.join(_path, 'index.pug'), html_vars, args.html_out)
+
+
+def write_html(template_file, vars, html_out):
+    """Given a template in `template_file`, format it in the Pug language
+    using `vars` as context variables, printing the resulting HTML to stdout.
+    """
+    with open(template_file) as f:
+        template = f.read()
+
+    parser = pypugjs.parser.Parser(template)
```

Comment:
It doesn't look like you're using pypugjs at all -- certainly can remove this boilerplate, as you're just shipping `dowker_faw.html`